apps/api/src/app/change/usecases/promote-feed-change/promote-feed-change.ts

```diff
@@ -0,0 +1,31 @@
+import { Injectable } from '@nestjs/common';
+import { PromoteTypeChangeCommand } from '../promote-type-change.command';
+import { FeedEntity, FeedRepository } from '@novu/dal';
+
+@Injectable()
+export class PromoteFeedChange {
+  constructor(private feedRepository: FeedRepository) {}
+
+  async execute(command: PromoteTypeChangeCommand) {
+    let item: FeedEntity | undefined = undefined;
+    if (command.item.name) {
+      item = await this.feedRepository.findOne({
+        _environmentId: command.environmentId,
+        _organizationId: command.organizationId,
+        name: command.item.name,
+      });
+    }
+
+    if (!item) {
+      return this.feedRepository.create({
+        name: command.item.name,
+        _environmentId: command.environmentId,
+        _organizationId: command.organizationId,
+      });
+    }
+
+    return await this.feedRepository.delete({
+      _id: item._id,
+    });
```

Comment:
@davidsoderberg I'm not sure i got it, why are we deleting the feed item here?

Reply:
because if we do not find the development env one we should delete prod one as well